templates/hooks/db_instance/sdk_update_post_set_output.go.tpl

```diff
@@ -1,8 +1,83 @@
-    // When ModifyDBInstance API is successful, it asynchronously
+	// ModifyDBInstance returns a DBInstance struct that contains the
+	// *previously set* values for various mutable fields. This is problematic
+	// because it causes a "flopping" behaviour when the user has modified a
+	// Spec field from value A to value B but the output shape from
+	// ModifyDBInstance for that field contains value A, the standard SetOutput
+	// Go code generated above will set the Spec field to the *old* value
+	// again. The next time the reconciler runs, it will attempt to modify the
+	// field from value B to value A again, causing a flop loop.
+	//
+	// Luckily, the Output shape's DBInstance struct contains a
+	// `PendingModifiedValues` struct which contains those field values that
+	// the user specified. So, we can use these to "reset" the Spec back to the
+	// appropriate user-specific values.
+	pmv := resp.DBInstance.PendingModifiedValues
+	if pmv != nil {
+		if pmv.AllocatedStorage != nil {
+			ko.Spec.AllocatedStorage = pmv.AllocatedStorage
+		}
+		//if pmv.AutomationMode != nil {
```

Comment:
Same question/request about adding a comment explaining why we're skipping these fields.

Reply:
@jljaco all done.